packages/terra-application-layout/src/menu/_ApplicationMenuConfigAdapter.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AppDelegate from 'terra-app-delegate';
+import { processedRoutesPropType } from 'terra-navigation-layout';
+import RoutingStackDelegate from 'terra-navigation-layout/lib/RoutingStackDelegate';
+
+import ApplicationMenu from './_ApplicationMenu';
+import ApplicationLayoutPropTypes from '../utils/propTypes';
+
+const propTypes = {
+  /**
+   * The AppDelegate instance to provide to the ApplicationMenu.
+   */
+  app: AppDelegate.propType,
+  /**
+   * Array of navigation Objects to be rendered as tabs within the Application Header.
+   */
+  navigationItems: ApplicationLayoutPropTypes.navigationItemsPropType,
+  /**
+   * The Object of layout-related APIs provided to the components of the Layout. This prop is provided by the Layout.
+   */
+  layoutConfig: ApplicationLayoutPropTypes.layoutConfigPropType.isRequired,
+  /**
+   * The set of routes currently identified by the NavigationLayout. This prop is provided by the NavigationLayout.
+   */
+  navigationLayoutRoutes: PropTypes.arrayOf(processedRoutesPropType),
+  /**
+   * The window size currently identified by the NavigationLayout. This prop is provided by the NavigationLayout.
+   */
+  navigationLayoutSize: PropTypes.string,
+  /**
+   * The Object containing RoutingStack APIs provided to children of the RoutingStack. This prop is provided by the RoutingStack.
+   */
+  routingStackDelegate: RoutingStackDelegate.propType,
+  /**
+   * The Object containing props from the configuration necessary for ApplicationMenu creation.
+   */
+  applicationMenuConfigAdapterProps: PropTypes.shape({
```

Comment:
Could we rename this prop to `applicationMenuProps`? Or possibly flatten these?

Reply:
I cannot flattening them exposes us to too much cross-talk between the content props, but I can rename it to something like `applicationMenuProps`